src/ERC20Permit.sol

```diff
@@ -83,7 +84,7 @@ abstract contract ERC20Permit is IERC20Permit, StatefulERC712 {
         uint256 spenderAllowance_ = allowance[sender_][msg.sender]; // Cache `spenderAllowance_` to stack.
 
         if (spenderAllowance_ != type(uint256).max) {
-            allowance[sender_][msg.sender] = spenderAllowance_ - amount_;
+            _approve(sender_, msg.sender, spenderAllowance_ - amount_);
```

Comment:
This will emit an `Approval` event. Is that standard/expected behaviour?

Reply:
Yes, it was working like this before, the integrations tests in TTG failed because the event was missing.
It allows applications to reconstruct the allowance for all accounts by listening to the event.
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L29